controllers/topics/broadcastMessage.js

```diff
@@ -42,7 +42,7 @@ const calculateDelay = (joinDate, delay) => {
 
 module.exports = async (req, res) => {
   const {userId} = req;
-  const {communityMessageFormatId, topicId, targetUserId, targetDate, delay} = req.body;
+  const {communityMessageFormatId, topicId, topicIds, targetUserId, targetDate, delay} = req.body;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `topicId` and `topicIds` variables seem to be redundant. It would be better to use only one of them to avoid confusion. If you want to support both single and multiple topic IDs, consider always using an array, even for a single ID. This way, the rest of your code can always assume it's working with an array, simplifying the logic.
```diff
-  const {communityMessageFormatId, topicId, topicIds, targetUserId, targetDate, delay} = req.body;
+  const {communityMessageFormatId, topicIds, targetUserId, targetDate, delay} = req.body;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->